src/Microsoft.Health.Dicom.SqlServer/Features/Schema/Migrations/2.sql

```diff
@@ -714,6 +781,158 @@ AS
     VALUES
         (@studyKey, @seriesKey, @instanceKey, @studyInstanceUid, @seriesInstanceUid, @sopInstanceUid, @newWatermark, @initialStatus, @currentDate, @currentDate)
 
+    -- Insert Custom Tags
+
+    -- String Key tags
+    IF EXISTS (SELECT 1 FROM @stringCustomTags)
+    BEGIN      
+        MERGE INTO dbo.CustomTagString AS T
+        USING 
+        (
+            SELECT input.TagKey, input.TagValue, input.TagLevel 
+            FROM @stringCustomTags input
+            INNER JOIN dbo.CustomTag WITH (REPEATABLEREAD) 
+            ON dbo.CustomTag.TagKey = input.TagKey
+            -- Not merge on custom tag which is being deleted.
+            AND dbo.CustomTag.TagStatus <> 2     
```

Comment:
Can you remind me all the TagStatus. I was thinking this would be only for Successfully provisioned/re-indexed tags. What does 0 and 1 mean?

Reply:
* 0: the tag is being reindexing
* 1: the tag has been added (reindex has been completed)
* 2: the tag is being deleted.
https://github.com/microsoft/dicom-server/blob/feature/customtags/src/Microsoft.Health.Dicom.Core/Features/CustomTag/CustomTagStatus.cs

Reply:
Thanks Peng. Condition makes sense